### DISCUSSION TITLE
Autocomplete

### DISCUSSION THREAD

**Discussion by beeinger** [sagiegurari/cargo-make#991] [Ideas]
Hey, been using cargo make and it's great, however I very much missed some autocomplete.
I realise it's not a subject for a PR to this repository, however I wanted to share this for anyone looking.

I'm currently using [Fig](https://fig.io/), so I made [this PR](https://github.com/withfig/autocomplete/pull/2180), will update you here if it gets merged.

When it gets merged you will be able to use Fig and get autocompletion on all available flags.
**Most importantly it autocompleted the custom tasks commands from the Makefile.toml**.
Resolving different locations of the makefile also works when the `--makefile <file>` flag is provided.

If anyone is interested in improving this lmk, also if there are any updates to the cargo-make API in the future, an update to Fig will be needed.

**Comment by sagiegurari** [comment]
thanks! that's great
1. i do have in this repo auto complete solutions such as bash and readme points to that
2. when your PR is keeged, can you do a PR for cargo make to document that?

**Reply by beeinger** [reply]
Sure! I didn't notice the Shell Completion chapter 👀
It's merged already so will make a PR here now.

**Reply by beeinger** [reply]
Done: https://github.com/sagiegurari/cargo-make/pull/993
Feel free to edit to your liking
